authentication/api.py

```diff
@@ -0,0 +1,150 @@
+"""API functions for authentication."""
+
+import logging
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from oauthlib.oauth2 import (
+    BackendApplicationClient,
+    InvalidGrantError,
+    TokenExpiredError,
+)
+from requests_oauthlib import OAuth2Session
+
+from authentication.models import KeycloakAdminToken
+from unified_ecommerce.celery import app
+
+User = get_user_model()
+log = logging.getLogger(__name__)
+
+
+def keycloak_session_init(url, **kwargs):
+    """
+    Initialize a Keycloak session.
+
+    This is a helper function that will initialize a Keycloak session with the
+    provided URL. It will also handle refreshing the token if it has expired.
+
+    Args:
+    url (str): The Keycloak admin URL.
+    **kwargs: Additional arguments to pass to the OAuth2Session initializer.
+
+    Returns:
+    None, or dict of data returned.
+    """
+
+    token_url = (
+        f"{settings.KEYCLOAK_ADMIN_URL}/auth/realms/master/"
+        "protocol/openid-connect/token"
+    )
+    client = BackendApplicationClient(client_id=settings.KEYCLOAK_ADMIN_CLIENT_ID)
+
+    auto_refresh_kwargs = {
+        "client_id": settings.KEYCLOAK_ADMIN_CLIENT_ID,
+        "client_secret": settings.KEYCLOAK_ADMIN_CLIENT_SECRET,
+    }
+
+    def update_token(token):
+        log_str = f"Refreshing Keycloak token {token}"
+        log.warning(log_str)
```

Comment:
Here and in a few other places `warning` is being used where `debug` is probably more appropriate. `warning` is more for log lines where something is going wrong but probably not as urgent as an error.